CA-409830: Confirm when overwriting an SR

rosslagerwall opened the pull request:
Add a confirmation dialog to confirm when the user selects a disk that contain an SR since it may result in data being lost.

It applies to the primary disk selection and the guest disk selection.

At the same time, drop the code related to "target-is-sr" since it did determine whether the target is an SR but didn't do anything with the information.

The dialog looks like this:

![sr](https://github.com/user-attachments/assets/af5fb576-5b8e-4f82-8a29-887dc3202c9c)

freddy77 commented on tui/installer/screens.py:
```diff
@@ -32,6 +32,12 @@
 
+sr_overwrite_msg = """The selected disk, {}, contains a storage repository. The storage repository may currently be used by other hosts.
+
+Selecting Yes will permanently erase the storage reposory and any virtual machine disks it contains.
```
Mentioned downstream: typo `erase the storage repository`

freddy77 commented on tui/installer/screens.py:
```diff
@@ -613,13 +613,20 @@ def select_primary_disk(answers):
+    # determine current usage
+    disk = diskutil.probeDisk(entry)
+    if disk.storage[0]:
+        val = snackutil.ButtonChoiceWindowEx(tui.screen,
+            "Confirm Disk Erasure",
+            sr_overwrite_msg.format(diskutil.getHumanDiskLabel(entry, short=True)),
+            ['Yes', 'No'], default=1)
```
Maybe writing a small utility function to reuse these lines? They are the same on below chunk.

freddy77 commented:
Why not targeting `master` branch first?

rosslagerwall commented:
The updated PR adds to commits to ensure the installer correctly identifies SRs.
It also addresses the comments and skip asking about the primary disk a second time when it is also used for the SR.

MarkSymsCtx commented on disktools.py:
```diff
@@ -185,6 +185,8 @@ def readAllInfo(self):
             if name.startswith('/dev/dm-'):
                 n = int(name[8:])
                 pv['pv_name'] = getDeviceMapperNode(n)
+            elif name.startswith('/dev/disk/by-id/dm-name-'):
```
The other option here, not sure if it is possible is to `realpath` the name of the PV.

rosslagerwall replied:
Possibly. That only works if what it is being compared against is also using the canonical path.

Deli has a PR here (https://github.com/xenserver/host-installer/pull/250) to fix this issue by comparing the two paths with `samepath()` since the LVM change affects more than just multipath. I'll probably take that change and drop this commit.

freddy77 commented on diskutil.py:
```diff
@@ -499,6 +503,7 @@ def __init__(self, device):
 INSTALL_RETAIL = 1
 STORAGE_LVM = 1
 STORAGE_OTHER = 2
+STORAGE_GFS2 = 2
```
This is conflicting with `STORAGE_OTHER`

rosslagerwall commented on diskutil.py:
```diff
@@ -537,23 +542,24 @@ def probeDisk(device):
+        possible_srs.add(device)
 
+    lv_tool = len(possible_srs) and LVMTool()
+    for part_device in possible_srs:
```
I've added a commit to address this (by raising an exception).

GeraldEV commented on diskutil.py:
```diff
@@ -537,23 +542,24 @@ def probeDisk(device):
                 disk.state = (True, part_device)
                 if num + 2 in tool.partitions:
```
Is this branch of the code still relevant? Could these SRs still persist on a modern installation? even one which was upgrade from the George release?
